Runtime/Shaders/MapsBaker.cs

```diff
@@ -0,0 +1,217 @@
+        List<TextureHolder> textureHolders,
+        Dictionary<int, MaterialPropertyData> materialProperties) {
+        if (!destMat.HasProperty(MeshSyncConstants._MaskMap)) {
+            return;
+        }
+
+        // Mask map:
+        // R: metal
+        // G: ao
+        // B: detail mask
+        // A: smoothness
+
+        bool texturesExist = false;
+
+        texturesExist |=
+            FindTexture(MeshSyncConstants._MetallicGlossMap, textureHolders, materialProperties,
+                MeshSyncConstants._Metallic, 0, out var metalTexture);
+
+        texturesExist |=
+            FindTexture(MeshSyncConstants._GlossMap, textureHolders, materialProperties,
+                MeshSyncConstants._Glossiness, 1, out var glossTexture);
+
+        // If there are no textures, don't bake anything, slider values can control everything:
+        if (!texturesExist) {
+            destMat.SetTextureSafe(MeshSyncConstants._MaskMap, null);
+            destMat.DisableKeyword(MeshSyncConstants._MASKMAP);
+            return;
+        }
+
+        var shader = LoadShader(SHADER_NAME_HDRP_MASK);
+        if (shader == null) {
+            return;
+        }
+
+        shader.SetTexture(SHADER_CONST_METALLIC, metalTexture.Texture);
+        shader.SetTexture(SHADER_CONST_SMOOTHNESS, glossTexture.Texture);
+
+        var texture = shader.RenderToTexture(destMat.GetTexture(MeshSyncConstants._MaskMap));
+
+        destMat.EnableKeyword(MeshSyncConstants._MASKMAP);
+
+        destMat.SetTextureSafe(MeshSyncConstants._MaskMap, texture);
+        
+        metalTexture.Dispose();
+        glossTexture.Dispose();
+    }
+#else
+    private static void BakeSmoothness(Material destMat,
+        List<TextureHolder> textureHolders,
+        Dictionary<int, MaterialPropertyData> materialProperties) {
+        if (!destMat.HasProperty(_SmoothnessTextureChannel)) {
+            return;
+        }
+
+        var smoothnessChannel = destMat.GetInt(_SmoothnessTextureChannel);
+
+        Texture2DDisposable rgbTexture = null;
+        int                 channelName;
+
+        bool texturesExist = false;
+
+        if (smoothnessChannel == 0) {
+            // Bake to metallic alpha
+            channelName = MeshSyncConstants._MetallicGlossMap;
+            texturesExist |=
+                FindTexture(MeshSyncConstants._MetallicGlossMap, textureHolders, materialProperties,
+                    MeshSyncConstants._Metallic, 0, out rgbTexture);
+        }
+        else if (smoothnessChannel == 1) {
+            // Bake to albedo alpha
+            channelName = MeshSyncConstants._BaseMap;
+            texturesExist |=
+                FindTexture(MeshSyncConstants._MainTex, textureHolders, materialProperties, 0, 0,
+                    out rgbTexture);
+        }
+        else {
+            Debug.LogError(
+                $"Unknown smoothness channel, cannot bake to option: {smoothnessChannel} set in the smoothness texture channel.");
```

Comment:
Most logs we have so far start with "[MeshSync]", so the user is aware where the error is coming from. Perhaps we could do the same here?

Reply:
Done